src/Illuminate/Http/Resources/TransformsToResource.php

```diff
@@ -32,11 +33,10 @@ public function toResource(?string $resourceClass = null): JsonResource
      */
     protected function guessResource(): JsonResource
     {
-        $className = get_class($this);
-
-        $resourceClass = static::guessResourceName();
-
-        throw_unless(class_exists($resourceClass), \LogicException::class, sprintf('Failed to find resource class for model [%s].', $className));
+        throw_unless(
+            class_exists($resourceClass = static::guessResourceName()),
+            LogicException::class, sprintf('Failed to find resource class for model [%s].', get_class($this))
+        );
 
         return $resourceClass::make($this);
```

Comment:
This might be a decent use case for `tap()`:
```suggestion
        $resourceClass = tap(static::guessResourceName(), fn (string $resourceClass) => throw_unless(
            class_exists($resourceClass),
            LogicException::class, sprintf('Failed to find resource class for model [%s].', get_class($this))
        ));

        return $resourceClass::make($this);
```

Reply:
Would be nice if one could do something like this:
```php
        $resourceClass = throw_unless(
            static::guessResourceName(),
            class_exists(...),
            LogicException::class, sprintf('Failed to find resource class for model [%s].', get_class($this))
        );

        return $resourceClass::make($this);
```

Reply:
Or if there was a global `when()` helper 🤔 
```php
        $resourceClass = when(
            static::guessResourceName(),
            class_exists(...),
            fn () => throw new LogicException(sprintf('Failed to find resource class for model [%s].', get_class($this))),
        );

        return $resourceClass::make($this);
```

Reply:
Or a `Stringable` helper 🤔 
```php
        $resourceClass = Str::of(static::guessResourceName())
            ->when(
                class_exists(...),
                fn () => throw new LogicException(sprintf('Failed to find resource class for model [%s].', get_class($this))),
            )
            ->value();

        return $resourceClass::make($this);
```